File: typescript/medium/reverse-words-in-a-string.ts
function reverseWords(s: string): string {
    const words = s.split(/\s+/); // split by spaces
    let res: string[] = [];

    for (let i = words.length - 1; i >= 0; i--) {
        res.push(words[i]);
        if (i !== 0) {
            res.push(" ");
        }
    }

    return res.join("");
}

File: csharp/medium/reverse-words-in-a-string.cs
string ReverseWords(string s)
{
    string[] words = s.Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
    List<string> res = new List<string>();

    for (int i = words.Length - 1; i >= 0; i--)
    {
        res.Add(words[i]);
        if (i != 0)
        {
            res.Add(" ");
        }
    }

    return string.Join("", res);
}
